src/main/java/net/fabricmc/fabric/impl/item/FuelRegistry.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, 2017, 2018 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.item;
+
+import it.unimi.dsi.fastutil.objects.Object2IntLinkedOpenHashMap;
+import it.unimi.dsi.fastutil.objects.Object2IntMap;
+import net.fabricmc.fabric.api.item.ItemPropertyRegistry;
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemProvider;
+import net.minecraft.tag.Tag;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Map;
+
+// TODO: Clamp values to 32767 (+ add hook for mods which extend the limit to disable the check?)
```

Comment:
I think adding in a comment or javadoc about why values are clamped (because of burn time in a furnace being saved/synced as a short) would help prevent a lot of issues asking why the warning comes up.

Reply:
Hell, why not just make it an `ItemPropertyRegistry<Short>` in the first place for the meantime?

Reply:
:thinking: 

That's a really good question! Though the values of fuel are stored as integers in the furnace, they're clamped primarily during server->client data sync - this is not hard to patch, most likely